homeassistant/components/neato/config_flow.py

```diff
@@ -1,112 +1,66 @@
-                    title=user_input[CONF_USERNAME],
-                    data=user_input,
-                    description_placeholders={"docs_url": DOCS_URL},
-                )
-
-        return self.async_show_form(
-            step_id="user",
-            data_schema=vol.Schema(
-                {
-                    vol.Required(CONF_USERNAME): str,
-                    vol.Required(CONF_PASSWORD): str,
-                    vol.Optional(CONF_VENDOR, default="neato"): vol.In(VALID_VENDORS),
-                }
-            ),
-            description_placeholders={"docs_url": DOCS_URL},
-            errors=errors,
-        )
-
-    async def async_step_import(self, user_input):
-        """Import a config flow from configuration."""
+    @property
+    def logger(self) -> logging.Logger:
+        """Return logger."""
+        return logging.getLogger(__name__)
 
-        if self._async_current_entries():
+    async def async_step_user(self, user_input: Optional[dict] = None) -> dict:
+        """Create an entry for the flow."""
+        current_entries = self._async_current_entries()
+        if current_entries and current_entries[0].version == self.VERSION:
+            # Already configured
             return self.async_abort(reason="already_configured")
 
-        username = user_input[CONF_USERNAME]
-        password = user_input[CONF_PASSWORD]
-        vendor = user_input[CONF_VENDOR]
-
-        error = await self.hass.async_add_executor_job(
-            self.try_login, username, password, vendor
-        )
-        if error is not None:
-            _LOGGER.error(error)
-            return self.async_abort(reason=error)
-
-        return self.async_create_entry(
-            title=f"{username} (from configuration)",
-            data={
-                CONF_USERNAME: username,
-                CONF_PASSWORD: password,
-                CONF_VENDOR: vendor,
-            },
-        )
+        return await super().async_step_user(user_input=user_input)
 
-    @staticmethod
-    def try_login(username, password, vendor):
-        """Try logging in to device and return any errors."""
-        this_vendor = None
-        if vendor == "vorwerk":
-            this_vendor = Vorwerk()
-        else:  # Neato
-            this_vendor = Neato()
+    async def async_step_reauth(self, data) -> dict:
+        """Perform reauth upon migration of old entries."""
+        return await self.async_step_reauth_confirm()
 
-        try:
-            Account(username, password, this_vendor)
-        except NeatoLoginException:
-            return "invalid_auth"
-        except NeatoRobotException:
-            return "unknown"
-
-        return None
+    async def async_step_reauth_confirm(
+        self, user_input: Optional[dict] = None
+    ) -> dict:
+        """Confirm reauth upon migration of old entries."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="reauth_confirm", data_schema=vol.Schema({})
```

Comment:
This should work. Please test it.

Reply:
Works like a charm :tada: Thanks

Reply:
I'm not following. Where do we call `show_form` twice?

Reply:
The commented line get called right after startup and again after the user click `reconfigure`. Then a form is shown. The user confirms and the function get called a third time but now with an empty dict.

Reply:
The frontend needs to continue the flow somehow, yes, when the user clicks reconfigure.